assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -41,10 +47,57 @@ interface Store {
 	ref?: HTMLElement;
 }
 
+interface Event {
+	keyCode: number;
+}
+
 type SelectorsStore = Pick< Store, 'context' | 'selectors' | 'ref' >;
 
 interactivityApiStore( {
 	state: {},
+	effects: {
+		woocommerce: {
+			keyboardAccess: ( store: Store ) => {
+				const { context, actions } = store;
+				let allowNavigation = true;
+
+				const handleKeyEvents = ( event: Event ) => {
+					if (
+						! allowNavigation ||
+						! context.woocommerce.isDialogOpen
```

Comment:
Should we use the optional chaining operator (?.) to safely access isDialogOpen? This prevents errors if context.woocommerce is undefined.